src/Illuminate/Support/ValidatedInput.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Illuminate\Support;
+
+use ArrayIterator;
+use Illuminate\Contracts\Support\ValidatedData;
+use stdClass;
+
+class ValidatedInput implements ValidatedData
+{
+    /**
+     * The underlying input.
+     *
+     * @var array
+     */
+    protected $input;
+
+    /**
+     * Create a new validated input container.
+     *
+     * @param  array  $input
+     * @return void
+     */
+    public function __construct(array $input)
+    {
+        $this->input = $input;
+    }
+
+    /**
+     * Get a subset containing the provided keys with values from the input data.
+     *
+     * @param  array|mixed  $keys
+     * @return array
+     */
+    public function only($keys)
+    {
+        $results = [];
+
+        $input = $this->input;
+
+        $placeholder = new stdClass;
+
+        foreach (is_array($keys) ? $keys : func_get_args() as $key) {
+            $value = data_get($input, $key, $placeholder);
+
+            if ($value !== $placeholder) {
+                Arr::set($results, $key, $value);
+            }
+        }
+
+        return $results;
+    }
+
+    /**
+     * Get all of the input except for a specified array of items.
+     *
+     * @param  array|mixed  $keys
+     * @return array
+     */
+    public function except($keys)
+    {
+        $keys = is_array($keys) ? $keys : func_get_args();
+
+        $results = $this->input;
+
+        Arr::forget($results, $keys);
+
+        return $results;
+    }
+
+    /**
+     * Get the input as a collection.
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    public function collect()
+    {
+        return new Collection($this->input);
+    }
```

Comment:
@ryangjchandler will appreciate this one

Reply:
This is beautiful.